packages/server/test/node.spec.ts

```diff
@@ -386,6 +375,129 @@ describe('Node Specific Cases', () => {
           expect(response.status).toBe(200);
           await expect(response.text()).resolves.toBe('Hello world!');
         });
+
+        it('responds with the correct status code', async () => {
+          const expected: { status: number; statusText: string }[] = [];
+          const received: { status: number; statusText: string; resText?: string }[] = [];
+          for (const statusCodeStr in STATUS_CODES) {
+            const status = Number(statusCodeStr);
+            if (
+              (serverImplName === 'fastify' && status === 204) ||
+              status === 205 ||
+              status === 304
+            ) {
+              // Fastify hangs on these responses
```

Comment:
This case does not only handle fastify, but also status code 205 and 304 in isolation.

Why does Fastify hang? Is it a bug in Fastify? Is it expected?

Reply:
I forgot to mark the PR as WIP. I was still experimenting stuff. Fixed it in the tests already